src/BizHawk.Client.Common/movie/import/IMovieImport.cs

```diff
@@ -20,21 +21,11 @@ public interface IMovieImport
 	internal abstract class MovieImporter : IMovieImport
 	{
 		protected const string EmulationOrigin = "emuOrigin";
-		protected const string Md5 = "MD5";
 		protected const string MovieOrigin = "MovieOrigin";
 
 		protected IDialogParent _dialogParent;
-
-		protected void MaybeSetCorePreference(string sysID, string coreName, string fileExt)
-		{
-			if (Config.PreferredCores[sysID] != coreName
-				&& _dialogParent.ModalMessageBox2(
-					$"{fileExt} movies will have a better chance of syncing using the {coreName} core. Change your core preference for {sysID} roms to {coreName} now?",
-					icon: EMsgBoxIcon.Question))
-			{
-				Config.PreferredCores[sysID] = coreName;
-			}
-		}
+		private IEmulator _emulator;
+		private delegate bool MatchesMovieHash(ReadOnlySpan<byte> romData);

```

Comment:
I don't really understand what that would accomplish; the delegate is very specific so I don't see any use of putting some generic form in `BizHawk.Common`. Also, what is a `RefStructFunc`?

Reply:
The thing I'm asking you to make (because it's missing from the standard library). I half-remember some limitation where type parameters aren't allowed to be `T: ref struct`, but if it is allowed, `public delegate TRet RefStructFunc<TParam, TRet>(TParam a) where TParam : ref struct => ...;`

Reply:
Yeah, type parameters cannot be of `ref struct` type: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/ref-struct

Reply:
;_;